utils_test.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// Check if expected results are in the registry
+func checkRegistryResults(expRes map[string]float64, mfs []*dto.MetricFamily, t *testing.T) {
+
+	res := make(map[string]float64)
+
+	if len(mfs) == 0 {
+		t.Fatal(fmt.Errorf("No metrics returned in registry"))
+	}
+
+	for i := range mfs {
+		res[mfs[i].GetName()] = mfs[i].Metric[0].GetGauge().GetValue()
+	}
+
+	for k, v := range expRes {
+		if _, ok := res[k]; !ok || res[k] != v {
+			t.Fatal(fmt.Errorf("Expected: %v: %v, got: %v: %v", k, v, k, res[k]))
```

Comment:
`res[k]` here can cause an exception if it's missing, use a variable created in the if